ReactQt/runtime/src/reactuimanager.cpp

```diff
@@ -111,43 +95,50 @@ void ReactUIManager::setChildren(int containerReactTag, const QList<int>& childr
     manageChildren(containerReactTag, QList<int>(), QList<int>(), childrenTags, indices, QList<int>());
 }
 
+//-----------------------------------------------------------------------------
+
+void ReactUIManager::removeChildren(QQuickItem* parent, const QList<int>& removeAtIndices) {
+
+    if (!removeAtIndices.isEmpty()) {
+
+        QList<QQuickItem*> itemsToRemove;
+        for (int i : removeAtIndices) {
+            itemsToRemove.push_back(parent->childItems().at(i));
+        }
+
+        for (QQuickItem* child : itemsToRemove) {
+            child->setParent(0);
+            m_views.remove(ReactAttachedProperties::get(child)->tag());
+            delete child;
```

Comment:
I had a bug - QQuickItem was removed from its parent when it shouldn't be. And I couldn't figure out when this happens because deleteLater() was used and stacktrace was unclear. So I substitute deleteLater() with direct deletion to prevent similar problems in future. But can get it back.

Reply:
@vkjr  I have found following so far: http://doc.qt.io/qt-5/qquickitem.html
```
Note: Use of QObject::deleteLater() to clean up graphics resources is not recommended as this will run at an arbitrary time and it is unknown if there will be an OpenGL context bound when the deletion takes place.
```
Seems better to follow your implementation :)